src/diffusers/loaders.py

```diff
@@ -2294,6 +2459,111 @@ def enable_lora_for_text_encoder(self, text_encoder: Optional[PreTrainedModel] =
             raise ValueError("Text Encoder not found.")
         set_adapter_layers(self.text_encoder, enabled=True)
 
+    def set_adapters(
+        self,
+        adapter_names: Union[List[str], str],
+        adapter_weights: List[float] = None,
+    ):
+        # Handle the UNET
+        self.unet.set_adapters(adapter_names, adapter_weights)
+
+        # Handle the Text Encoder
+        if hasattr(self, "text_encoder"):
+            self.set_adapters_for_text_encoder(adapter_names, self.text_encoder, adapter_weights)
+        if hasattr(self, "text_encoder_2"):
+            self.set_adapters_for_text_encoder(adapter_names, self.text_encoder_2, adapter_weights)
+
+    def disable_lora(self):
+        if not self.use_peft_backend:
+            raise ValueError("PEFT backend is required for this method.")
+
+        # Disable unet adapters
+        self.unet.disable_lora()
+
+        # Disable text encoder adapters
+        if hasattr(self, "text_encoder"):
+            self.disable_lora_for_text_encoder(self.text_encoder)
+        if hasattr(self, "text_encoder_2"):
+            self.disable_lora_for_text_encoder(self.text_encoder_2)
+
+    def enable_lora(self):
+        if not self.use_peft_backend:
+            raise ValueError("PEFT backend is required for this method.")
+
+        # Enable unet adapters
+        self.unet.enable_lora()
+
+        # Enable text encoder adapters
+        if hasattr(self, "text_encoder"):
+            self.enable_lora_for_text_encoder(self.text_encoder)
+        if hasattr(self, "text_encoder_2"):
+            self.enable_lora_for_text_encoder(self.text_encoder_2)
+
+    def get_active_adapters(self) -> Optional[List[str]]:
+        """
+        Gets the list of the current active adapters
+        """
+        if not self.use_peft_backend:
+            raise ValueError(
+                "PEFT backend is required for this method. Please install the latest version of PEFT `pip install -U peft`"
+            )
+
+        from peft.tuners.tuners_utils import BaseTunerLayer
+
+        active_adapters = None
+
+        for module in self.unet.modules():
+            if isinstance(module, BaseTunerLayer):
+                active_adapters = module.active_adapters
+                break
+
+        return active_adapters
+
+    def set_lora_device(self, adapter_names: List[str], device: Union[torch.device, str, int]) -> None:
+        """
+        Sets the LoRAs that are listed in `adapter_names` into a target device. Usefull for offloading the LoRA in CPU
+        in case one wants to load multiple adapters and free some GPU memory.
+
+        Note this is specific to LoRA and needs some tweaking for new adapter architectures (e.g. IA3, AdaLora..)
```

Comment:
```suggestion
        Sets the LoRAs listed in `adapter_names` into a target device. Useful for offloading the LoRA in the CPU
        in case you want to load multiple adapters and free some GPU memory.

        This is specific to LoRA and needs some tweaking for new adapter architectures (IA3, AdaLora).
```